src/megaapi_impl.cpp

```diff
@@ -1495,6 +1495,128 @@ char *MegaApiImpl::getBlockedPath()
 }
 #endif
 
+void MegaApiImpl::backupFolder(const char *localFolder, const char *backupName, MegaRequestListener *listener)
+{
+    MegaRequestPrivate *request = new MegaRequestPrivate(MegaRequest::TYPE_BACKUP_FOLDER, listener);
+    request->setNodeHandle(0);
+    request->setFile(localFolder);
+    request->setName(backupName);
+
+    requestQueue.push(request);
+    waiter->notify();  // followup in MegaApiImpl::backupFolder_sendPendingRequest()
+}
+
+error MegaApiImpl::backupFolder_sendPendingRequest(MegaRequestPrivate* request) // request created in MegaApiImpl::backupFolder()
+{
+    // validate local path and backup name
+    string localPath(request->getFile());
+    string backupName(request->getName());
+
+    if (backupName.empty()) // get the last leaf of local path
+    {
+        // trim trailing path separator(s)
+        while (!localPath.empty() && (localPath.back() == '\\' || localPath.back() == '/'))  localPath.pop_back();
+
+        // find the last non-trailing separator
+        auto sep = localPath.find_last_of("\\/");
```

Comment:
We should reuse the logic from `LocalPath` here. Please, see:
- `LocalPath LocalPath::fromPath()`
- `LocalPath LocalPath::leafName()`
- `string LocalPath::toPath()`